src/app/pages/PerpetualPage/contexts/PerpetualQueriesContext.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { createContext, useEffect, useState } from 'react';
+import {
+  initialAmmState,
+  usePerpetual_queryAmmState,
+} from '../hooks/usePerpetual_queryAmmState';
+import {
+  initialLiqPoolState,
+  usePerpetual_queryLiqPoolStateFromPerpetualId,
+} from '../hooks/usePerpetual_queryLiqPoolStateFromPerpetualId';
+import {
+  initialPerpParameters,
+  usePerpetual_queryPerpParameters,
+} from '../hooks/usePerpetual_queryPerpParameters';
+import {
+  initialTraderState,
+  usePerpetual_queryTraderState,
+} from '../hooks/usePerpetual_queryTraderState';
+import {
+  AMMState,
+  LiqPoolState,
+  PerpParameters,
+  TraderState,
+} from '../utils/perpUtils';
+
+type PerpetualQueriesContextValue = {
+  ammState: AMMState;
+  perpetualParameters: PerpParameters;
+  traderState: TraderState;
+  liquidityPoolState: LiqPoolState;
+};
+
+export const PerpetualQueriesContext = createContext<
+  PerpetualQueriesContextValue
+>({
+  ammState: initialAmmState,
+  perpetualParameters: initialPerpParameters,
+  traderState: initialTraderState,
+  liquidityPoolState: initialLiqPoolState,
+});
+
+export const PerpetualQueriesContextProvider: React.FC = ({ children }) => {
+  const [value, setValue] = useState<PerpetualQueriesContextValue>({
+    ammState: initialAmmState,
+    perpetualParameters: initialPerpParameters,
+    traderState: initialTraderState,
+    liquidityPoolState: initialLiqPoolState,
+  });
+
+  const ammState = usePerpetual_queryAmmState();
+  const perpetualParameters = usePerpetual_queryPerpParameters();
+  const traderState = usePerpetual_queryTraderState();
+  const liquidityPoolState = usePerpetual_queryLiqPoolStateFromPerpetualId();
+
+  useEffect(
+    () =>
+      setValue({
+        ammState,
+        perpetualParameters,
+        traderState,
+        liquidityPoolState,
+      }),
+    [ammState, liquidityPoolState, perpetualParameters, traderState],
+  );
```

Comment:
The state and effect combination is not needed here, you can just use useMemo.
with effect the provider needs to be called twice by react.